src/precice/impl/ReadDataContext.cpp

```diff
@@ -0,0 +1,23 @@
+#include "ReadDataContext.hpp"
+
+namespace precice {
+namespace impl {
+ReadDataContext::ReadDataContext(
+    mesh::PtrData data,
+    mesh::PtrMesh mesh)
+    : DataContext(data, mesh)
+{
+}
+
+void ReadDataContext::configureMapping(MappingContext mappingContext, MeshContext meshContext)
+{
+  PRECICE_TRACE();
+  PRECICE_ASSERT(meshContext.mesh->hasDataName(getDataName()));
+  mesh::PtrData data = meshContext.mesh->data(getDataName());
+  PRECICE_ASSERT(data != _providedData);
+  this->setMapping(mappingContext, data, _providedData);
```

Comment:
We shouldn't need to use this explicitly.
```suggestion
  setMapping(mappingContext, data, _providedData);
```